trouble with multiple CTE's

`WITH a as (SELECT ...), b as (SELECT * FROM a) SELECT * FROM b` generates a panic:

```
$ make sql
sqlc generate
panic: relation "queued_job_id" does not exist

goroutine 1 [running]:
github.com/kyleconroy/sqlc/internal/dinosql.NewQueryCatalog(0xc0001f2e10, 0xc000483c50, 0x1, 0x1)
	/Users/kevin/src/github.com/kyleconroy/sqlc/internal/dinosql/parser.go:628 +0x2e8
github.com/kyleconroy/sqlc/internal/dinosql.sourceTables(0xc0001f2e10, 0x46840a0, 0xc000426690, 0x4, 0x4681b20, 0x33d, 0x4, 0xb)
	/Users/kevin/src/github.com/kyleconroy/sqlc/internal/dinosql/parser.go:692 +0x1fd
github.com/kyleconroy/sqlc/internal/dinosql.outputColumns(0xc0001f2e10, 0x46840a0, 0xc000426690, 0x4, 0xc0005353e0, 0x1, 0x1, 0xc0001d0680)
	/Users/kevin/src/github.com/kyleconroy/sqlc/internal/dinosql/parser.go:729 +0x73
github.com/kyleconroy/sqlc/internal/dinosql.parseQuery(0xc0001f2e10, 0x4683820, 0xc0004a5d00, 0xc0003d7500, 0x967, 0xc0001d0600, 0x8, 0x10)
	/Users/kevin/src/github.com/kyleconroy/sqlc/internal/dinosql/parser.go:445 +0x50a
github.com/kyleconroy/sqlc/internal/dinosql.ParseQueries(0xc0001f2e10, 0xc000232430, 0x9, 0x45778a6, 0xa, 0xc000232440, 0x9, 0xc000232460, 0x10, 0xc000232450, ...)
	/Users/kevin/src/github.com/kyleconroy/sqlc/internal/dinosql/parser.go:240 +0x636
github.com/kyleconroy/sqlc/internal/cmd.glob..func4(0x4a6c8c0, 0x4aa2b40, 0x0, 0x0)
	/Users/kevin/src/github.com/kyleconroy/sqlc/internal/cmd/cmd.go:169 +0x105a
github.com/spf13/cobra.(*Command).execute(0x4a6c8c0, 0x4aa2b40, 0x0, 0x0, 0x4a6c8c0, 0x4aa2b40)
	/Users/kevin/src/github.com/spf13/cobra/command.go:833 +0x29d
github.com/spf13/cobra.(*Command).ExecuteC(0xc0001dca00, 0xc0000b5f10, 0x1, 0x1)
	/Users/kevin/src/github.com/spf13/cobra/command.go:917 +0x2fb
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/kevin/src/github.com/spf13/cobra/command.go:867
github.com/kyleconroy/sqlc/internal/cmd.Do(0xc00000e090, 0x1, 0x1, 0x467c960, 0xc000010010, 0x467c980, 0xc000010018, 0x467c980, 0xc000010020, 0xc00008e058)
	/Users/kevin/src/github.com/kyleconroy/sqlc/internal/cmd/cmd.go:35 +0x240
main.main()
	/Users/kevin/src/github.com/kyleconroy/sqlc/cmd/sqlc/main.go:10 +0xad
make: *** [Makefile:84: sql] Error 2
```

--- a/internal/dinosql/parser.go
+++ b/internal/dinosql/parser.go
@@ -453,11 +453,15 @@ func parseQuery(c core.Catalog, stmt nodes.Node, source string) (*Query, error) 
 		return nil, err
 	}
 
-	cols, err := outputColumns(c, raw.Stmt)
+	qc, err := buildQueryCatalog(c, raw.Stmt)
 	if err != nil {
 		return nil, err
 	}
-	expanded, err := expand(c, raw, rawSQL)
+	cols, err := outputColumns(qc, raw.Stmt)
+	if err != nil {
+		return nil, err
+	}
+	expanded, err := expand(qc, raw, rawSQL)
 	if err != nil {
 		return nil, err
 	}
@@ -499,7 +503,7 @@ type edit struct {
 	New      string
 }
 
-func expand(c core.Catalog, raw nodes.RawStmt, sql string) (string, error) {
+func expand(qc *QueryCatalog, raw nodes.RawStmt, sql string) (string, error) {
 	list := search(raw, func(node nodes.Node) bool {
 		switch node.(type) {
 		case nodes.DeleteStmt:
@@ -516,7 +520,7 @@ func expand(c core.Catalog, raw nodes.RawStmt, sql string) (string, error) {
 	}
 	var edits []edit
 	for _, item := range list.Items {
-		edit, err := expandStmt(c, raw, item)
+		edit, err := expandStmt(qc, raw, item)
 		if err != nil {
 			return "", err
 		}
@@ -525,8 +529,8 @@ func expand(c core.Catalog, raw nodes.RawStmt, sql string) (string, error) {
 	return editQuery(sql, edits)
 }
 
-func expandStmt(c core.Catalog, raw nodes.RawStmt, node nodes.Node) ([]edit, error) {
-	tables, err := sourceTables(c, node)
+func expandStmt(qc *QueryCatalog, raw nodes.RawStmt, node nodes.Node) ([]edit, error) {
+	tables, err := sourceTables(qc, node)
 	if err != nil {
 		return nil, err
 	}
@@ -629,23 +633,32 @@ type QueryCatalog struct {
 	ctes    map[string]core.Table
 }
 
-func NewQueryCatalog(c core.Catalog, with *nodes.WithClause) QueryCatalog {
-	ctes := map[string]core.Table{}
+func buildQueryCatalog(c core.Catalog, node nodes.Node) (*QueryCatalog, error) {
+	var with *nodes.WithClause
+	switch n := node.(type) {
+	case nodes.UpdateStmt:
+		with = n.WithClause
+	case nodes.SelectStmt:
+		with = n.WithClause
+	default:
+		with = nil
+	}
+	qc := &QueryCatalog{catalog: c, ctes: map[string]core.Table{}}
 	if with != nil {
 		for _, item := range with.Ctes.Items {
 			if cte, ok := item.(nodes.CommonTableExpr); ok {
-				cols, err := outputColumns(c, cte.Ctequery)
+				cols, err := outputColumns(qc, cte.Ctequery)
 				if err != nil {
-					panic(err.Error())
+					return nil, err
 				}
-				ctes[*cte.Ctename] = core.Table{
+				qc.ctes[*cte.Ctename] = core.Table{
 					Name:    *cte.Ctename,
 					Columns: cols,
 				}
 			}
 		}
 	}
-	return QueryCatalog{catalog: c, ctes: ctes}
+	return qc, nil
 }
 
 func (qc QueryCatalog) GetTable(fqn core.FQN) (core.Table, *core.Error) {
@@ -673,9 +686,8 @@ func (qc QueryCatalog) GetTable(fqn core.FQN) (core.Table, *core.Error) {
 // Return an error if column references don't exist
 // Return an error if a table is referenced twice
 // Return an error if an unknown column is referenced
-func sourceTables(c core.Catalog, node nodes.Node) ([]core.Table, error) {
+func sourceTables(qc *QueryCatalog, node nodes.Node) ([]core.Table, error) {
 	var list nodes.List
-	var with *nodes.WithClause
 	switch n := node.(type) {
 	case nodes.DeleteStmt:
 		list = nodes.List{
@@ -686,12 +698,10 @@ func sourceTables(c core.Catalog, node nodes.Node) ([]core.Table, error) {
 			Items: []nodes.Node{*n.Relation},
 		}
 	case nodes.UpdateStmt:
-		with = n.WithClause
 		list = nodes.List{
 			Items: append(n.FromClause.Items, *n.Relation),
 		}
 	case nodes.SelectStmt:
-		with = n.WithClause
 		list = search(n.FromClause, func(node nodes.Node) bool {
 			_, ok := node.(nodes.RangeVar)
 			return ok
@@ -699,8 +709,6 @@ func sourceTables(c core.Catalog, node nodes.Node) ([]core.Table, error) {
 	default:
 		return nil, fmt.Errorf("sourceTables: unsupported node type: %T", n)
 	}
-
-	qc := NewQueryCatalog(c, with)
 
 	var tables []core.Table
 	for _, item := range list.Items {
@@ -736,8 +744,8 @@ func HasStarRef(cf nodes.ColumnRef) bool {
 //
 // Return an error if column references are ambiguous
 // Return an error if column references don't exist
-func outputColumns(c core.Catalog, node nodes.Node) ([]core.Column, error) {
-	tables, err := sourceTables(c, node)
+func outputColumns(qc *QueryCatalog, node nodes.Node) ([]core.Column, error) {
+	tables, err := sourceTables(qc, node)
 	if err != nil {
 		return nil, err
 	}
@@ -865,7 +873,7 @@ func outputColumns(c core.Catalog, node nodes.Node) ([]core.Column, error) {
 				name = *res.Name
 			}
 
-			fun, err := c.LookupFunctionN(fqn, len(n.Args.Items))
+			fun, err := qc.catalog.LookupFunctionN(fqn, len(n.Args.Items))
 			if err == nil {
 				cols = append(cols, core.Column{Name: name, DataType: fun.ReturnType, NotNull: true})
 			} else {

--- a/internal/endtoend/testdata/star_expansion_cte/go/query.sql.go
+++ b/internal/endtoend/testdata/star_expansion_cte/go/query.sql.go
@@ -5,6 +5,7 @@ package querytest
 
 import (
 	"context"
+	"database/sql"
 )
 
 const starExpansionCTE = `-- name: StarExpansionCTE :many
@@ -21,6 +22,42 @@ func (q *Queries) StarExpansionCTE(ctx context.Context) ([]Bar, error) {
 	for rows.Next() {
 		var i Bar
 		if err := rows.Scan(&i.C, &i.D); err != nil {
+			return nil, err
+		}
+		items = append(items, i)
+	}
+	if err := rows.Close(); err != nil {
+		return nil, err
+	}
+	if err := rows.Err(); err != nil {
+		return nil, err
+	}
+	return items, nil
+}
+
+const starExpansionTwoCTE = `-- name: StarExpansionTwoCTE :many
+WITH 
+  a AS (SELECT a, b FROM foo),
+  b AS (SELECT 1::int as bar, a, b FROM a)
+SELECT bar, a, b FROM b
+`
+
+type StarExpansionTwoCTERow struct {
+	Bar int32
+	A   sql.NullString
+	B   sql.NullString
+}
+
+func (q *Queries) StarExpansionTwoCTE(ctx context.Context) ([]StarExpansionTwoCTERow, error) {
+	rows, err := q.db.QueryContext(ctx, starExpansionTwoCTE)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+	var items []StarExpansionTwoCTERow
+	for rows.Next() {
+		var i StarExpansionTwoCTERow
+		if err := rows.Scan(&i.Bar, &i.A, &i.B); err != nil {
 			return nil, err
 		}
 		items = append(items, i)

--- a/internal/endtoend/testdata/star_expansion_cte/query.sql
+++ b/internal/endtoend/testdata/star_expansion_cte/query.sql
@@ -1,4 +1,11 @@
 CREATE TABLE foo (a text, b text);
 CREATE TABLE bar (c text, d text);
+
 -- name: StarExpansionCTE :many
 WITH cte AS (SELECT * FROM foo) SELECT * FROM bar;
+
+-- name: StarExpansionTwoCTE :many
+WITH 
+  a AS (SELECT * FROM foo),
+  b AS (SELECT 1::int as bar, * FROM a)
+SELECT * FROM b;

